app/main.py

```diff
@@ -1,2 +1,45 @@
+import typing
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __setitem__(self, key: typing.Hashable, value: typing.Any) -> None:
+        if self.length > self.capacity * 2 / 3:
+            self.resize()
+        index = hash(key) % self.capacity
+        if self.hash_table[index]:
+            for item in self.hash_table[index]:
+                if item[0] == key and hash(item[0]) == hash(key):
+                    item[1] = value
+                    return
+        self.hash_table[index].append([key, value])
+        self.length += 1
```

Comment:
It doesn't work with collision when we have different keys with the same hash, use a while loop and change index when the cell is not empty. Also, it is better to add a hash with the key and value in the list.